src/nvim/event/process.c

```diff
@@ -221,29 +221,15 @@ void process_stop(Process *proc) FUNC_ATTR_NONNULL_ALL
   }
   proc->stopped_time = os_hrtime();
 
-  switch (proc->type) {
-    case kProcessTypeUv:
-      // Close the process's stdin. If the process doesn't close its own
-      // stdout/stderr, they will be closed when it exits(possibly due to being
-      // terminated after a timeout)
-      stream_may_close(&proc->in);
-      os_proc_tree_kill(proc->pid, SIGTERM);
```

Comment:
This "close stdin" step was from aa9cb48bf08af14068178619, before we fixed/reworked the SIGTERM timing logic. So it's probably outdated / no longer needed, as you guessed in 7933eb46b4c5b14bad7fd1bb735713fb13988d13 ; but the commit message should explain this.